File: JavaScriptPractice/Leetcode303.js
var theSumList=[0]
var NumArray = function(nums) 
{   
    let theSum = 0;
    for (let i = 0; i < nums.length; i++) {
        theSum += nums[i]
        theSumList.push(theSum)
    }

};

NumArray.prototype.sumRange = function(left, right) 
{
    return theSumList[right+1]-theSumList[left]
};

File: CSharpPractice/LeetcodePractice/Leetcode303.cs
public class NumArray {

    public List<int> theSumList = new List<int>(0);
    public NumArray(int[] nums) {
        int theSum = 0;
        for (var i = 0; i < nums.Count(); i++)
        {
            theSum +=nums[i];
            theSumList.Add(theSum);   
        }   
    }
    public int SumRange(int left, int right) {
        
        return theSumList[right+1] - theSumList[left];
    }
}
